### DISCUSSION TITLE
CDN IP Scanner

### DISCUSSION THREAD

**Discussion by vfarid** [2dust/v2rayNG#2970] [Ideas]
I recentry added a feature to scan CDN clean IPs and sent a PR but @2dust rejected the PR said it could be done in a seperate app.
Imo, if we want to test IPs using real delay method, the other app need to have almost all features of v2rayNG in order to test clean IPs using real configs. Furtheremore we can update related configs with found clean IPs in the future.
For untrained users, this is the best way to simplify the whole process (Find and apply clean IPs) and since this feature is very light and almost have no affect on general app performance, i think its better to re-think about it.

**Comment by 2dust** [comment]
The CDN is provided by a commercial company
Such a feature should not be added

**Reply by vfarid** [reply]
@2dust I investigated a lot and read many docs from cloudflare (the main target for scanner). This kind of IP scanner is not against their rules. If you are not happy with its name, we can name it in another way like "Find best IP".
What i did in scanner is to replace the the config address with some random IP of Cloudflare CDN and test the config with real ping algorithm, nothig is againt the rules and regulations.
As i know, this feature will be usefull for both China and Iran.
If you let me add this feature, you will see it would be very usefull for all users specially for those who know less about how to to improve configs.
Thank you.
